nomad/core_sched_test.go

```diff
@@ -2160,103 +2032,103 @@ func TestAllocation_GCEligible(t *testing.T) {
-			ThresholdIndex: 90,
-			ShouldGC:       false,
+			Desc:          "Don't GC when non terminal",
+			ClientStatus:  structs.AllocClientStatusPending,
+			DesiredStatus: structs.AllocDesiredStatusRun,
+			GCTime:        now,
+			ModifyTime:    now.UnixNano(),
+			CutoffTime:    now,
+			ShouldGC:      false,
 		},
 		{
-			Desc:           "Don't GC when non terminal and job stopped",
-			ClientStatus:   structs.AllocClientStatusPending,
-			DesiredStatus:  structs.AllocDesiredStatusRun,
-			JobStop:        true,
-			GCTime:         fail,
-			ModifyIndex:    90,
-			ThresholdIndex: 90,
-			ShouldGC:       false,
+			Desc:          "Don't GC when non terminal and job stopped",
+			ClientStatus:  structs.AllocClientStatusPending,
+			DesiredStatus: structs.AllocDesiredStatusRun,
+			JobStop:       true,
+			GCTime:        now,
+			ModifyTime:    now.UnixNano(),
+			CutoffTime:    now,
+			ShouldGC:      false,
 		},
 		{
-			Desc:           "Don't GC when non terminal and job dead",
-			ClientStatus:   structs.AllocClientStatusPending,
-			DesiredStatus:  structs.AllocDesiredStatusRun,
-			JobStatus:      structs.JobStatusDead,
-			GCTime:         fail,
-			ModifyIndex:    90,
-			ThresholdIndex: 90,
-			ShouldGC:       false,
+			Desc:          "Don't GC when non terminal and job dead",
+			ClientStatus:  structs.AllocClientStatusPending,
+			DesiredStatus: structs.AllocDesiredStatusRun,
+			JobStatus:     structs.JobStatusDead,
+			GCTime:        now,
+			ModifyTime:    now.UnixNano(),
+			CutoffTime:    now,
+			ShouldGC:      false,
 		},
 		{
-			Desc:           "Don't GC when non terminal on client and job dead",
-			ClientStatus:   structs.AllocClientStatusRunning,
-			DesiredStatus:  structs.AllocDesiredStatusStop,
-			JobStatus:      structs.JobStatusDead,
-			GCTime:         fail,
-			ModifyIndex:    90,
-			ThresholdIndex: 90,
-			ShouldGC:       false,
+			Desc:          "Don't GC when non terminal on client and job dead",
+			ClientStatus:  structs.AllocClientStatusRunning,
+			DesiredStatus: structs.AllocDesiredStatusStop,
+			JobStatus:     structs.JobStatusDead,
+			GCTime:        now,
+			ModifyTime:    now.UnixNano(),
+			CutoffTime:    now,
+			ShouldGC:      false,
 		},
 		{
 			Desc:             "GC when terminal but not failed ",
 			ClientStatus:     structs.AllocClientStatusComplete,
 			DesiredStatus:    structs.AllocDesiredStatusRun,
-			GCTime:           fail,
-			ModifyIndex:      90,
-			ThresholdIndex:   90,
+			GCTime:           now,
+			ModifyTime:       now.UnixNano(),
+			CutoffTime:       now,
 			ReschedulePolicy: nil,
 			ShouldGC:         true,
 		},
 		{
 			Desc:             "Don't GC when threshold not met",
 			ClientStatus:     structs.AllocClientStatusComplete,
 			DesiredStatus:    structs.AllocDesiredStatusStop,
-			GCTime:           fail,
-			ModifyIndex:      100,
-			ThresholdIndex:   90,
+			GCTime:           now,
+			ModifyTime:       now.UnixNano(),
+			CutoffTime:       now.Add(-1 * time.Hour),
```

Comment:
Oh hey! I think this is susceptible to daylight savings time bugs! Don't change it to half an hour either as that could still break if @philrenaud runs the tests at just the wrong time while visiting Newfoundland: https://en.wikipedia.org/wiki/Newfoundland_Time_Zone (I may enjoy timezone hijinks too much.)